App/Talent.App.WebApp/wwwroot/js/react/ReactScripts/Profile/ContactDetail.jsx

```diff
@@ -2,269 +2,296 @@
+          maxLength={80}
+          placeholder="Enter an email"
+          errorMessage="Please enter a valid email"
+        />
+
+        <ChildSingleInput
+          inputType="text"
+          label="Phone number"
+          name="phone"
+          value={this.state.newContact.phone}
+          controlFunc={this.handleChange}
+          maxLength={12}
+          placeholder="Enter a phone number"
+          errorMessage="Please enter a valid phone number"
+        />
+
+        <button type="button" className="ui teal button" onClick={this.saveContact}>Save</button>
+        <button type="button" className="ui button" onClick={this.closeEdit}>Cancel</button>
+      </div>
+    )
+  }
+
+  renderDisplay() {
+
+    let fullName = this.props.details ? `${this.props.details.firstName} ${this.props.details.lastName}` : ""
+    let email = this.props.details ? this.props.details.email : ""
+    let phone = this.props.details ? this.props.details.phone : ""
+
+    return (
+      <div className='row'>
+        <div className="ui sixteen wide column">
+          <React.Fragment>
+            <p>Name: {fullName}</p>
+            <p>Email: {email}</p>
+            <p>Phone: {phone}</p>
+          </React.Fragment>
+          <button type="button" className="ui right floated teal button" onClick={this.openEdit}>Edit</button>
+        </div>
+      </div>
+    )
+  }
 }
 
 
 export class CompanyDetailSection extends Component {
-    constructor(props) {
-        super(props)
-
-        const details = props.details ?
-            Object.assign({}, props.details)
-            : {
-                name: "",
-                email: "",
-                phone: ""
-            }
-
-        this.state = {
-            showEditSection: false,
-            newContact: details
-        }
-
-        this.openEdit = this.openEdit.bind(this)
-        this.closeEdit = this.closeEdit.bind(this)
-        this.handleChange = this.handleChange.bind(this)
-        this.saveContact = this.saveContact.bind(this)
-        this.renderEdit = this.renderEdit.bind(this)
-        this.renderDisplay = this.renderDisplay.bind(this)
-    }
+  constructor(props) {
+    super(props)
 
-    openEdit() {
-        const details = Object.assign({}, this.props.details)
-        this.setState({
-            showEditSection: true,
-            newContact: details
-        })
-    }
+    const details = props.details ?
+      Object.assign({}, props.details)
+      : {
+        name: "",
+        email: "",
+        phone: ""
+      }
 
-    closeEdit() {
-        this.setState({
-            showEditSection: false
-        })
+    this.state = {
+      showEditSection: false,
+      newContact: details
     }
+    this.schema = Yup.object().shape({
+      name: Yup.string().required('Name required'),
+      email: Yup.string().email('Email is not valid').required('Email required'),
+      phone: Yup.string().matches(TalentUtil.phoneRegExp(), 'Phone number is not valid'),
+    });
```

Comment:
Yup is a very helpful library for form validation. But as you can see here, you are setting up the same fields again and again. Instead of this, you can create a single global Yup schema and call it wherever you need it so that you will no longer need to set up multiple fields with the same name, format, and validation errors